Sources/UIApplication+handleSDLEvents.swift

```diff
@@ -93,29 +94,29 @@ extension UIApplication {
 
 extension UIApplication {
     static func onWillEnterForeground() {
-        UIApplication.restart {
-            if let runningApplication = UIApplication.shared {
-                runningApplication.delegate?.applicationWillEnterForeground(runningApplication)
-            }
+        #if os(Android)
+        if UIScreen.main == nil { // sometimes we "enter foreground" after just a loss of focus
+            UIScreen.main = UIScreen()
         }
+        #endif
+
+        UIApplication.shared?.delegate?.applicationWillEnterForeground(UIApplication.shared)
     }
 
     static func onDidEnterForeground() {
-        if let runningApplication = UIApplication.shared {
-            runningApplication.delegate?.applicationDidBecomeActive(runningApplication)
-        }
+        UIApplication.shared?.delegate?.applicationDidBecomeActive(UIApplication.shared)
     }
 
     static func onWillEnterBackground() {
-        if let runningApplication = UIApplication.shared {
-            runningApplication.delegate?.applicationWillResignActive(runningApplication)
-        }
+        UIApplication.shared?.delegate?.applicationWillResignActive(UIApplication.shared)
     }
 
     static func onDidEnterBackground() {
-        if let runningApplication = UIApplication.shared {
-            runningApplication.delegate?.applicationDidEnterBackground(runningApplication)
-        }
+        UIApplication.shared?.delegate?.applicationDidEnterBackground(UIApplication.shared)
+
+        #if os(Android)
+        UIScreen.main = nil
```

Comment:
woulnd't it make sense to remove the instance for mac, too?

Reply:
We could I guess but it's not necessary